drivers/ethernet/eth_native_posix.c

```diff
@@ -356,6 +363,35 @@ static void eth_iface_init(struct net_if *iface)
 	net_if_set_link_addr(iface, ll_addr->addr, ll_addr->len,
 			     NET_LINK_ETHERNET);
 
+#ifdef CONFIG_NET_IPV4
+	if (ipv4_addr_cmd_opt != NULL) {
+		if (net_addr_pton(AF_INET, ipv4_addr_cmd_opt, &addr)) {
+			NET_ERR("Invalid address: %s", ipv4_addr_cmd_opt);
+			return;
```

Comment:
I'm not sure, should those errors really be fatal? Perhaps it'd be good to proceed with interface initialization even if the user provides invalid IP address?